lib/ansible/plugins/shell/powershell.py

```diff
@@ -1046,8 +1046,10 @@ def remove(self, path, recurse=False):
 
     def mkdtemp(self, basefile, system=False, mode=None, tmpdir=None):
         basefile = self._escape(self._unquote(basefile))
-        # FIXME: Support system temp path and passed in tmpdir!
-        return self._encode_script('''(New-Item -Type Directory -Path $env:temp -Name "%s").FullName | Write-Host -Separator '';''' % basefile)
+        if tmpdir is None:
+            tmpdir = "$env:temp"
+
+        return self._encode_script('''(New-Item -Type Directory -Path %s -Name "%s").FullName | Write-Host -Separator '';''' % (tmpdir, basefile))
```

Comment:
What happens if tmpdir consists of a path with spaces in it ?
I expect Powershell to fail on this, an it should be quoted `...Directory -Path "%s" -Name..."